src/NuGet.Clients/NuGet.SolutionRestoreManager/SolutionRestoreJob.cs

```diff
@@ -179,12 +168,28 @@ internal sealed class SolutionRestoreJob : ISolutionRestoreJob
                     .OfType<IDependencyGraphProject>()
                     .ToList();
 
-                await RestorePackageSpecProjectsAsync(
-                    dependencyGraphProjects,
-                    forceRestore,
-                    isSolutionAvailable,
-                    restoreSource,
-                    token);
+                // Only continue if there are some build integrated type projects.
+                if (projects.Any(project => project is BuildIntegratedNuGetProject))
+                {
+                    if (solutionDirectory == null)
```

Comment:
This changes the behavior for `packages.config` based solutions, is this intentional?

Reply:
can you please elaborate on the change in behavior?

Reply:
The code you mentioned checks for `isSolutionAvailable` whereas `SolutionDirectory` is still null and that code will just return without logging any error and continue with restore.

Reply:
I have added the check to PC based projects as well.

Reply:
Instead of duplicating the check, wouldn't it be better to check for if there's any project in `projects` list and `SolutionDIrectory` is `null` then log this error and return? Move this check after line153